components/apps_form/apps_form_field/apps_form_select_field.tsx

```diff
@@ -73,6 +88,21 @@ export default class AppsFormSelectField extends React.PureComponent<Props, Stat
         return this.props.performLookup(this.props.field.name, userInput);
     }
 
+    loadDynamicUserOptions = async (userInput: string): Promise<AppSelectOption[]> => {
+        const usersSearchResults: UserAutocomplete = await this.props.actions.autocompleteUsers(userInput.toLowerCase());
+        return usersSearchResults.users.map((user) => ({...user, label: this.props.teammateNameDisplay ? displayUsername(user, this.props.teammateNameDisplay) : user.username, value: user.id, icon_data: Client4.getUsersRoute() + '/' + user.id + '/image?_=' + (user.last_picture_update || 0)}));
```

Comment:
For the icon data, can this be computed on it's own line into a local variable, for better readability? Also I'm thinking there may be a more established way in the project that fetching the avatar is done

Reply:
This would be much easier to read if these were done step-by-step in local variables.

Also, I see this is indeed how it was computed in the previous component:

https://github.com/mattermost/mattermost-webapp/blob/3417f0eb2d02efd5f935c8f02963cf8b144d7ccc/components/suggestion/generic_user_provider.tsx#L56

I think we can use `imageURLForUser` which is being used in `select_user_option`